inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp

```diff
@@ -0,0 +1,164 @@
+
+#include <ngraph/pass/visualize_tree.hpp>
+#include <low_precision/fake_quantize.hpp>
+
+#include "common_test_utils/ngraph_test_utils.hpp"
+#include "simple_low_precision_transformer.hpp"
+
+using namespace testing;
+using namespace ngraph;
+using namespace ngraph::pass;
+
+class FakeQuantizeWithDynamicIntervalsTransformationTestValues {
+public:
+    low_precision::LayerTransformation::Params params;
+    bool inputLowConst;
+    bool inpuHighConst;
+    bool outputLowConst;
+    bool outputHighConst;
+};
+
+inline std::ostream& operator<<(std::ostream& os, const std::vector<float>& values) {
+    os << "{ ";
+    for (size_t i = 0; i < values.size(); ++i) {
+        os << values[i];
+        if (i != (values.size() - 1ul)) {
+            os << ", ";
+        }
+    }
+    os << " }";
+    return os;
+}
+
+inline std::ostream& operator<<(std::ostream& out, const FakeQuantizeWithDynamicIntervalsTransformationTestValues& testValue) {
+    return out << "_" <<
+        testValue.inputLowConst << "_" <<
+        testValue.inpuHighConst << "_" <<
+        testValue.outputLowConst << "_" <<
+        testValue.outputHighConst;
+}
+
+typedef std::tuple<
+    ngraph::element::Type,
+    ngraph::Shape,
+    FakeQuantizeWithDynamicIntervalsTransformationTestValues> FakeQuantizeTransformationParams;
+
+class FakeQuantizeWithDynamicIntervalsTransformation : public LayerTransformation, public testing::WithParamInterface<FakeQuantizeTransformationParams> {
+public:
+    void SetUp() override {
+        const ngraph::element::Type precision = std::get<0>(GetParam());
+        const ngraph::Shape shape = std::get<1>(GetParam());
+        const FakeQuantizeWithDynamicIntervalsTransformationTestValues testValues = std::get<2>(GetParam());
+
+        actualFunction = get(precision, shape, testValues.inputLowConst, testValues.inpuHighConst, testValues.outputLowConst, testValues.outputHighConst);
+
+        SimpleLowPrecisionTransformer transform;
+        transform.add<ngraph::pass::low_precision::FakeQuantizeTransformation, ngraph::opset1::FakeQuantize>(testValues.params);
+        transform.transform(actualFunction);
+
+        referenceFunction = get(precision, shape, testValues.inputLowConst, testValues.inpuHighConst, testValues.outputLowConst, testValues.outputHighConst);
+    }
+
+    static std::string getTestCaseName(testing::TestParamInfo<FakeQuantizeTransformationParams> obj) {
+        ngraph::element::Type precision;
+        ngraph::Shape shape;
+        FakeQuantizeWithDynamicIntervalsTransformationTestValues testValues;
+        std::tie(precision, shape, testValues) = obj.param;
+
+        std::ostringstream result;
+        result << LayerTransformation::getTestCaseNameByParams(precision, shape, testValues.params) << testValues;
+        return result.str();
+    }
+
+private:
+    std::shared_ptr<ngraph::Node> makeTranspose(const ngraph::element::Type precision, const ngraph::Shape shape, const std::vector<float> low) {
```

Comment:
Looks like this is common stuff and should be moved to ngraph builders.

Reply:
Motivations to have private method: method size is small (3 lines), simple and not configurable. There are custom predefined values (precision, shape, transpose constant value). If move to common part then input arguments list will be huge.

Reply:
refactored, input node is used